tests/cpp/CMakeLists.txt

```diff
@@ -14,8 +14,38 @@ if(USE_CUDA)
 endif()
 
 file(GLOB_RECURSE SYCL_TEST_SOURCES "plugin/test_sycl_*.cc")
-if(NOT PLUGIN_SYCL)
-  list(REMOVE_ITEM TEST_SOURCES ${SYCL_TEST_SOURCES})
+list(REMOVE_ITEM TEST_SOURCES ${SYCL_TEST_SOURCES})
+
+if(PLUGIN_SYCL)
+  set(CMAKE_CXX_COMPILER "icpx")
+  file(GLOB_RECURSE SYCL_TEST_SOURCES "plugin/test_sycl_*.cc")
+  add_library(plugin_sycl_test OBJECT ${SYCL_TEST_SOURCES})
+
+  target_include_directories(plugin_sycl_test
+    PRIVATE
+    ${gtest_SOURCE_DIR}/include
+    ${xgboost_SOURCE_DIR}/include
+    ${xgboost_SOURCE_DIR}/dmlc-core/include
+    ${xgboost_SOURCE_DIR}/rabit/include)
+
+  target_compile_definitions(plugin_sycl_test PUBLIC -DXGBOOST_USE_SYCL=1)
+
+  target_link_libraries(plugin_sycl_test PUBLIC -fsycl)
+
+  set_target_properties(plugin_sycl_test PROPERTIES
+  COMPILE_FLAGS -fsycl
```

Comment:
should we also have "-fsycl-device-code-split=per_kernel" as options?

Reply:
> should we also have "-fsycl-device-code-split=per_kernel" as options?

I didn't observe any perf differences, at least for the last compiler.